enterprise/internal/database/perms_store.go

```diff
@@ -2042,3 +2047,168 @@ func sliceToSet[T comparable](s []T) map[T]struct{} {
+			return rows.Scan(
+				&r.ID,
+				&r.Name,
+				&r.Private,
+				&dbutil.NullString{S: &r.ExternalRepo.ID},
+				&dbutil.NullString{S: &r.ExternalRepo.ServiceType},
+				&dbutil.NullString{S: &r.ExternalRepo.ServiceID},
+				&dbutil.NullString{S: &r.URI},
+				&dbutil.NullString{S: &r.Description},
+				&r.Fork,
+				&r.Archived,
+				&dbutil.NullInt{N: &r.Stars},
+				&r.CreatedAt,
+				&dbutil.NullTime{Time: &r.UpdatedAt},
+				&dbutil.NullTime{Time: &r.DeletedAt},
+				metadata,
+				blocked,
+				kvps,
+				sources,
+				&dbutil.NullTime{Time: &updatedAt},
+				&reason,
+			)
+		})
+
+		if err := scanRepo(s.logger, rows, &repo); err != nil {
+			return nil, err
+		}
+
+		// If authz is bypassed due to user being site-admin then show permissions
+		// reason as `UserRepoPermissionReasonSiteAdmin`.
+		if authzParams.BypassAuthzReason == database.BypassAuthzReasonSiteAdmin {
+			reason = UserRepoPermissionReasonSiteAdmin
+		}
+
+		perms = append(perms, &UserPermission{Repo: &repo, Reason: reason, UpdatedAt: updatedAt})
+	}
+
+	return perms, nil
+}
+
+const reposPermissionInfoQueryFmt = `
+WITH accessible_repos AS (
+	SELECT
+		repo.id,
+		repo.name,
+		repo.private,
+		repo.external_id,
+		repo.external_service_type,
+		repo.external_service_id,
+		repo.uri,
+		repo.description,
+		repo.fork,
+		repo.archived,
+		repo.stars,
+		repo.created_at,
+		repo.updated_at,
+		repo.deleted_at,
+		repo.metadata,
+		repo.blocked,
+		(SELECT json_object_agg(key, value) FROM repo_kvps WHERE repo_kvps.repo_id = repo.id),
+		(
+			SELECT	json_agg(json_build_object('CloneURL', esr.clone_url, 'ID', esr.external_service_id, 'Kind', LOWER(svcs.kind)))
+			FROM external_service_repos AS esr
+			JOIN external_services AS svcs ON esr.external_service_id = svcs.id
+			WHERE esr.repo_id = repo.id AND svcs.deleted_at IS NULL
+		)
+	FROM repo
+	WHERE 
+		repo.deleted_at IS NULL 
+		AND %s -- Authz Conds, Pagination Conds, Search
+	ORDER BY %s
+	%s -- Limit
+)
+
+SELECT
+	ar.*,
+	up.updated_at AS permission_updated_at,
+	CASE 
+		WHEN up.user_id IS NOT NULL THEN 'Permission Sync' 
+		ELSE 'Unrestricted' -- If no user_permissions entry is found then the accessible repo must be unrestricted
+	END AS permission_reason
+FROM
+	accessible_repos AS ar
+	LEFT JOIN user_permissions AS up ON up.user_id = %d
+		AND up.object_ids_ints @> INTSET (ar.id)
+`
+
+type UserPermission struct {
+	Repo      *types.Repo
+	Reason    UserRepoPermissionReason
+	UpdatedAt time.Time
+}
+
+type UserRepoPermissionReason string
+
+// UserRepoPermissionReason constants.
+const (
+	UserRepoPermissionReasonAuthzBypass    UserRepoPermissionReason = "Authentication Bypassed"
+	UserRepoPermissionReasonSiteAdmin      UserRepoPermissionReason = "Site Admin"
+	UserRepoPermissionReasonUnrestricted   UserRepoPermissionReason = "Unrestricted"
+	UserRepoPermissionReasonPermissionSync UserRepoPermissionReason = "Permission Sync"
```

Comment:
we're doing plural from some time back :)
```suggestion
	UserRepoPermissionReasonPermissionSync UserRepoPermissionReason = "Permissions Sync"
```